core/controllers/components/ApisystemComponent.php

```diff
@@ -404,13 +408,25 @@ public function uploadGeneratetoken($args)
                     MIDAS_POLICY_READ
                 )
                 ) {
+                    $create_additional_revision = isset($args['create_additional_revision']) ? $args['create_additional_revision'] : false;
                     $revision = $itemModel->getLastRevision($item);
 
-                    if ($revision === false) {
-                        // Create new revision if none exists yet
+                    // do not create an additional revision if last revision has one bitstream and checkum matches
+                    if ($create_additional_revision && $revision) {
+                      $bitstreams = $revision->getBitstreams();
+                      if (count($bitstreams) == 1 && $args['checksum'] == $bitstreams[0]->getChecksum()) {
+                        return array('token' => '');
```

Comment:
Returning an empty token is valid and documented, this is what is done in the rest of the code. 
See https://github.com/midasplatform/Midas/blob/fec6bda441a385d8f14642d6644e5b291177c493/core/controllers/components/ApisystemComponent.php#L332